Drivers/Comms/Src/Xbee.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Xbee.cpp
+ *
+ *  Created on: Aug 20, 2022
+ *      Author: Aaditya
+ */
+
+#include "Xbee.hpp"
+#include "main.h"
+#include "usart.h"
+
+#define UART_TIMEOUT 100
+
+#define DRONE_BASIC_PAYLOAD_TYPE 1
+
+void changeEndian(uint8_t* val, int size) {
+	uint8_t temp;
+	for (int x = 0; x < size/2; x++) {
+		temp = val[x];
+		val[x] = val[size - 1 - x];
+		val[size - 1 - x] = temp;
+	}
+}
+
+void TransmitMessage(DroneBasicPayload* payload) {
+	Packet packet = {};
+	TransmitFrameHeader tf = {};
+
+	packet.time = HAL_GetTick();
+	packet.payloadType = DRONE_BASIC_PAYLOAD_TYPE;
+
+	uint8_t checksum = 0xFF;
+
+	uint8_t rawPacket[sizeof(TransmitFrameHeader) + sizeof(Packet) + sizeof(DroneBasicPayload) + sizeof(checksum)];
+
+
+	uint8_t* tfArr = (uint8_t*)&tf;
+	uint8_t* packetArr = (uint8_t*)&packet;
+	uint8_t* payloadArr = (uint8_t*)payload;
+
+	tf.messageLength = sizeof(TransmitFrameHeader) + sizeof(Packet) + sizeof(DroneBasicPayload) - 3;
```

Comment:
Why -3 on `tf.messageLength`, is it a magic number???

Reply:
yeah, right now the driver only sends data from zp to ground, because we need it for IMU data (for PID tuning).
the - 3 is because the xbee wants message length to be the length of the message without including the start deliminator, which is 1 byte, and the 2 bytes for the length. I can leave a comment explaining why the -3 exists, but I dont think it makes sense to make it a constant or anything tbh, I wouldnt know what to call it.

Reply:
I might just call it "packet offset" and add the comment, I don't know if we want magic words or magic numbers